### PR TITLE
feat: billing address

### PR DESCRIPTION
**Pull request by tobi-or-not-tobi**
The billing address can be used to add or select a specific address. When there's no address list (default during anonymous checkout), the shipping address can be reused. 

closes [HRZ-2587](https://spryker.atlassian.net/browse/HRZ-2587)

Breaking changes
- renamed checkout delivery component to checkout shipping address
- auto select the default shipping address rather than the first address
- billing address is added to checkout steps

[HRZ-2587]: https://spryker.atlassian.net/browse/HRZ-2587?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/domain/checkout/billing-address/billing-address.component.ts [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+@hydratable()
+export class CheckoutBillingAddressComponent
+  extends CheckoutMixin(LitElement)
+  implements isValid
+{
+  static styles = [checkoutBillingAddressStyles];
+
+  protected addressService = resolve(AddressService);
+
+  protected addresses = signal(this.addressService.getAddresses());
```
Comment (Sinucid) [inline libs/domain/checkout/billing-address/billing-address.component.ts]:
```suggestion
  protected $addresses = signal(this.addressService.getAddresses());
```

File: libs/domain/checkout/billing-address/billing-address.component.ts [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  protected selected = signal(this.checkoutStateService.get('billingAddress'));
```
Comment (Sinucid) [inline libs/domain/checkout/billing-address/billing-address.component.ts]:
```suggestion
  protected $selected = signal(this.checkoutStateService.get('billingAddress'));
```

File: libs/domain/checkout/billing-address/billing-address.component.ts [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+  protected shippingAddress = signal(
```
Comment (Sinucid) [inline libs/domain/checkout/billing-address/billing-address.component.ts]:
```suggestion
  protected $shippingAddress = signal(
```

File: libs/domain/checkout/billing-address/billing-address.component.ts [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    this.checkoutStateService.get('shippingAddress')
+  );
+
+  protected sameAsShippingAddress = signal(true);
```
Comment (Sinucid) [inline libs/domain/checkout/billing-address/billing-address.component.ts]:
```suggestion
  protected $isSameAsShippingAddress = signal(true);
```

File: libs/domain/checkout/billing-address/billing-address.component.ts [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+
+  @query('oryx-checkout-address')
+  protected checkoutAddress?: CheckoutAddressComponent;
+
+  protected override render(): TemplateResult {
+    return html`${this.renderHeading()}${this.renderBody()}`;
+  }
+
+  protected renderHeading(): TemplateResult[] {
+    const templates = [
```
Comment (Sinucid) [inline libs/domain/checkout/billing-address/billing-address.component.ts]: Why that better then this?

```
return html`
      <h3>${i18n('checkout.steps.billing-address')}</h3>
      ${when(
        this.addresses()?.length,
        () => html`<oryx-checkout-manage-address
          @change=${this.onChange}
          .selected=${this.selected()}
        ></oryx-checkout-manage-address>`,
        () => html`<oryx-button .type=${ButtonType.Text}>
          <button @click=${this.reuseShippingAddress}>
            ${i18n(
              this.sameAsShippingAddress()
                ? 'checkout.billing-address.change'
                : 'checkout.billing-address.same-as-shipping-address'
            )}
          </button>
        </oryx-button>`
      )}`;
```

File: libs/domain/checkout/billing-address/billing-address.component.ts [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+  protected reuseShippingAddress(): void {
+    this.sameAsShippingAddress.set(!this.sameAsShippingAddress());
+  }
+
+  isValid(report: boolean): boolean {
+    return !!this.checkoutAddress?.isValid(report);
+  }
+
+  protected onChange(e: CustomEvent<AddressEventDetail>): void {
+    if (!this.sameAsShippingAddress() || this.addresses()?.length) {
```
Comment (Sinucid) [inline libs/domain/checkout/billing-address/billing-address.component.ts]:
```
 if (e.detail?.address && (!this.sameAsShippingAddress() || this.addresses()?.length)) {
      this.persist(e.detail.address, e.detail.valid);
}
```

File: libs/domain/checkout/manage-address/manage-address.component.ts [hunk 7]
```diff
@@ -161,6 +147,7 @@ export class CheckoutManageAddressComponent extends AddressMixin(
 
   protected onClose(): void {
     this.addressStateService.clear();
+    this.preselected = undefined;
```
Comment (Sinucid) [inline libs/domain/checkout/manage-address/manage-address.component.ts]: `this.preselected = null;`
and default value == null

**Review by Sinucid — CHANGES_REQUESTED** [review]

File: libs/domain/checkout/billing-address/billing-address.component.ts (hunk 5 above)
Comment (tobi-or-not-tobi) [reply libs/domain/checkout/billing-address/billing-address.component.ts]: fair enough, I roll back to that

**Review by Sinucid — DISMISSED** [review]

**Review by Sinucid — APPROVED** [review]

**Pull request merged**